crates/ruff_python_formatter/src/statement/stmt_try.rs

```diff
@@ -1,13 +1,175 @@
+                        dynamic_text(name.as_str(), None)
+                    ]
+                )?;
+            }
+        }
+        write!(f, [text(":"), block_indent(&body.format())])?;
+
+        dangling_node_comments(except_handler_except_handler).fmt(f)?;
+        trailing_node_comments(except_handler_except_handler).fmt(f)
+    }
+}
+
+impl<'ast> AsFormat<PyFormatContext<'ast>> for ExceptHandler {
+    type Format<'a> = FormatRefWithRule<
+        'a,
+        ExceptHandler,
+        FormatExceptHandler,
+        PyFormatContext<'ast>,
+    > where Self: 'a;
+
+    fn format(&self) -> Self::Format<'_> {
+        FormatRefWithRule::new(self, FormatExceptHandler::default())
+    }
+}
+
 impl FormatNodeRule<StmtTry> for FormatStmtTry {
     fn fmt_fields(&self, item: &StmtTry, f: &mut PyFormatter) -> FormatResult<()> {
-        write!(f, [not_yet_implemented(item)])
+        let StmtTry {
+            range: _,
+            body,
+            handlers,
+            orelse,
+            finalbody,
+        } = item;
+
+        let comments_info = f.context().comments().clone();
+        let mut dangling_comments = comments_info.dangling_comments(item.as_any_node_ref());
+
+        write!(
+            f,
+            [
+                text("try:"),
+                hard_line_break(),
+                block_indent(&body.format()),
+            ]
+        )?;
+
+        for handler in handlers {
+            let handler_comments_start =
+                dangling_comments.partition_point(|comment| comment.slice().end() <= handler.end());
+            let (handler_comments, rest) = dangling_comments.split_at(handler_comments_start);
+            dangling_comments = rest;
+
+            if lines_before(
+                handler_comments
+                    .first()
+                    .map_or(handler.range().start(), |c| c.slice().start()),
+                f.context().contents(),
+            ) > 1
+            {
+                write!(f, [empty_line()])?;
+            }
+
+            write!(f, [leading_comments(handler_comments), &handler.format()])?;
+        }
+
+        if let [.., last] = &orelse[..] {
+            let orelse_comments_start =
+                dangling_comments.partition_point(|comment| comment.slice().end() <= last.end());
+            let (orelse_comments, rest) = dangling_comments.split_at(orelse_comments_start);
+            dangling_comments = rest;
+            if let Some(first_orelse_comment) = orelse_comments.first() {
+                if lines_before(first_orelse_comment.slice().start(), f.context().contents()) > 1 {
+                    write!(f, [empty_line()])?;
+                }
+            }
+            write!(
+                f,
+                [
+                    leading_comments(orelse_comments),
+                    text("else:"),
+                    block_indent(&orelse.format())
+                ]
+            )?;
+        }
+        if let [.., last] = &finalbody[..] {
```

Comment:
Nit: The logic for formatting the handlers,finally, and orelse seem somewhat repetitive. Could we split some of it into a function `format_case(name: &str, body: &Suite, dangling_comments: &[SourceComment]) -> FormatResult<&[SourceComment]>`? 

Feel free to follow up on this in a separate PR or adding a todo (you can add my name). I'm more than happy merging this as it is

Reply:
i've got something working for this (separate pr coming). a bit messier than the above since we need different handling of the handlers compared to orelse/finally (i think)